### PR TITLE
Interactive MSAL UI tests

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: MSAL/test/automation/tests/MSALADFSBaseUITest.m [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+- (NSString *)runSharedADFSInteractiveLoginWithRequest:(MSALTestRequest *)request
+                                       closeResultView:(BOOL)closeResultView
+{
+    // 1. Do interactive login
+    NSDictionary *config = [self configWithTestRequest:request];
+    [self acquireToken:config];
+
+    if (!request.useEmbedded
+        && !request.useSFController)
```
Comment (unpluggedk) [inline MSAL/test/automation/tests/MSALADFSBaseUITest.m]: if we are to have these type of booleans, can we just have useAuthSession?
or have the type as a variable?

File: MSAL/test/automation/tests/MSALBaseAADUITest.m [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+#pragma mark - Shared parameterized steps
+
+- (NSString *)runSharedAADLoginWithTestRequest:(MSALTestRequest *)request
+{
+    NSDictionary *config = [self configWithTestRequest:request];
+    [self acquireToken:config];
+
+    if (!request.useSFController
+        && !request.useEmbedded
+        && !request.usePassedWebView)
```
Comment (unpluggedk) [inline MSAL/test/automation/tests/MSALBaseAADUITest.m]: same as before, can we have more straight forward check?

File: MSAL/test/automation/tests/MSALBaseUITest.h [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+@interface MSALBaseUITest : XCTestCase
+
+@property (nonatomic) XCUIApplication *testApp;
+@property (nonatomic, class) MSIDTestAccountsProvider *accountsProvider;
+@property (nonatomic) MSIDTestAccount *primaryAccount;
+@property (nonatomic) MSIDTestAutomationConfiguration *testConfiguration;
+
+- (void)assertRefreshTokenInvalidated;
+- (void)assertAccessTokenExpired;
+- (void)assertAuthUIAppearWithEmbedded:(BOOL)embedded safariViewController:(BOOL)safariViewController;
```
Comment (unpluggedk) [inline MSAL/test/automation/tests/MSALBaseUITest.h]: name is a little confusing, perhaps change it to something like
assertAuthUIAppearWithWebview:type

or simply separate this into different methods?

File: MSAL/test/automation/tests/MSALBaseUITest.h [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+- (void)clearCache;
+- (void)clearKeychain;
+- (void)clearCookies;
+- (void)aadEnterEmail:(NSString *)email;
+- (void)aadEnterEmail;
+- (void)aadEnterEmailInApp:(XCUIApplication *)app;
+- (void)aadEnterPassword;
+- (void)aadEnterPassword:(NSString *)password;
+- (void)aadEnterPasswordInApp:(XCUIApplication *)app;
+- (void)aadEnterPassword:(NSString *)password testApp:(XCUIApplication *)testApp;
```
Comment (unpluggedk) [inline MSAL/test/automation/tests/MSALBaseUITest.h]: Q: do we ever really call this function? is this for multi app?

File: MSAL/test/automation/tests/MSALBaseUITest.h [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+- (void)adfsEnterPassword;
+- (void)adfsEnterPasswordInApp:(XCUIApplication *)app;
+- (void)adfsEnterPassword:(NSString *)password;
+- (void)adfsEnterPassword:(NSString *)password testApp:(XCUIApplication *)testApp;
+- (void)acceptMSSTSConsentIfNecessary:(NSString *)acceptButtonTitle embeddedWebView:(BOOL)embeddedWebView;
+- (void)closeAuthUIWithEmbedded:(BOOL)embedded safariViewController:(BOOL)safariViewController;
```
Comment (unpluggedk) [inline MSAL/test/automation/tests/MSALBaseUITest.h]: same as above, either a different name or separate methods?

File: MSAL/test/automation/tests/MSALTestRequest.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+@property (nonatomic, strong) NSString *redirectUri;
+@property (nonatomic, strong) NSString *authority;
+@property (nonatomic, strong) NSString *cacheAuthority;
+@property (nonatomic, strong) NSString *uiBehavior;
+@property (nonatomic, strong) NSString *accountIdentifier;
+@property (nonatomic, strong) NSString *loginHint;
+@property (nonatomic, strong) MSIDTestAccount *testAccount;
+@property (nonatomic) BOOL useEmbedded;
+@property (nonatomic) BOOL useSFController;
+@property (nonatomic) BOOL usePassedWebView;
```
Comment (unpluggedk) [inline MSAL/test/automation/tests/MSALTestRequest.h]: Can we combine these like discussed above?

File: MSAL/test/automation/tests/MSALADFSBaseUITest.m (hunk 1 above)
Comment (oldalton) [reply MSAL/test/automation/tests/MSALADFSBaseUITest.m]: Fixed that one after our conversation yesterday :)

File: MSAL/test/automation/ios/MSALAutomationConstants.m [hunk 7]
```diff
@@ -28,13 +28,15 @@
 NSString *const MSAL_VALIDATE_AUTHORITY_PARAM                   = @"validate_authority";
 NSString *const MSAL_CLIENT_ID_PARAM                            = @"client_id";
 NSString *const MSAL_AUTHORITY_PARAM                            = @"authority";
-NSString *const MSAL_USER_IDENTIFIER_PARAM                      = @"user_identifier";
+NSString *const MSAL_ACCOUNT_IDENTIFIER_PARAM                      = @"home_account_identifier";
```
[the remark this answers is not in the document]
Comment (oldalton) [reply MSAL/test/automation/ios/MSALAutomationConstants.m]: fixed

File: MSAL/test/automation/tests/MSALBaseUITest.h (hunk 3 above)
Comment (oldalton) [reply MSAL/test/automation/tests/MSALBaseUITest.h]: fixed

File: MSAL/test/automation/tests/MSALBaseUITest.h (hunk 4 above)
Comment (oldalton) [reply MSAL/test/automation/tests/MSALBaseUITest.h]: yes, multiapp needs this

File: MSAL/test/automation/tests/MSALBaseUITest.h (hunk 5 above)
Comment (oldalton) [reply MSAL/test/automation/tests/MSALBaseUITest.h]: fixed :)

File: MSAL/test/automation/tests/MSALTestRequest.h (hunk 6 above)
Comment (oldalton) [reply MSAL/test/automation/tests/MSALTestRequest.h]: yes, done :)

File: MSAL/test/automation/tests/MSALBaseUITest.m [hunk 8]
```diff
@@ -293,7 +293,10 @@ - (void)closeAuthUIUsingEmbeddedWebview:(BOOL)embedded safariViewController:(BOO
 {
     NSString *buttonTitle = @"Cancel";
 
-    if (safariViewController)
+    BOOL usesSafariAsFallback = ([[[UIDevice currentDevice] systemVersion] floatValue] <= 11.0f
+                                 && !embedded);
+
```
Comment (unpluggedk) [inline MSAL/test/automation/tests/MSALBaseUITest.m]: Should version check be less than 11, not less than or equal to?

btw, Why not just pass webview type to this method and handle it within?
i.e./
if (webviewtype == sfv ||
    (webviewtype == default && os < 11) buttontitle = done.

Comment (oldalton) [reply MSAL/test/automation/tests/MSALBaseUITest.m]: Done :)

**Pull request merged**